### PR TITLE
[sdb] Add Module.ApplyChanges

### PR DESCRIPTION
**Pull request by lambdageek**
Add ApplyChanges command to Mono.Debugger.Soft

Related mono/debugger-libs PR https://github.com/mono/debugger-libs/pull/342
Related dotnet/runtime PR https://github.com/dotnet/runtime/pull/49043

Contributes to https://github.com/dotnet/runtime/issues/44806

### CODE REVIEW

**Review by thaystg — APPROVED** [review]

**Review by vargaz — APPROVED** [review]

**Comment by lambdageek** [comment]
@akoeplinger  "mono-ci (Linux x64 All Profiles Build)" correctly says there's an apidiff change.  Not sure if monojenkins commit command still works, or if there's something else now.

**Comment by lambdageek** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins build failed

File: mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/ModuleMirror.cs [hunk 1]
```diff
@@ -70,5 +70,13 @@ public class ModuleMirror : Mirror
 				return info.SourceLink;
 			}
 		}
+
+		// Apply a hot reload delta to the current module
+		// Since protocol version 2.60
+		public void ApplyChanges (ArrayMirror dmeta, ArrayMirror dIL, Value dPDB) {
+		    /* dPDB is Value because it can be ArrayMirror or PrimitiveValue (vm, null) */
+		    vm.CheckProtocolVersion (2, 60);
```
Comment (jbevain) [inline mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/ModuleMirror.cs:78]: You're not bumping the protocol version supported by the client.

**Pull request merged**